### ISSUE TITLE
IndexOutOfRangeException when constructing RecurrencePattern

### ISSUE THREAD

**Issue by derekantrican** [rianjs/ical.net#550]
The following throws an IndexOutOfRangeException:

```csharp
new RecurrencePattern("RRULE:FREQ=WEEKLY;BYDAY=MO;UNTIL=20220708T110000Z;INTERVAL=2;");
```

After a small amount of testing (which is by no means conclusive), I think the issue is with the trailing semicolon. Removing that does not throw an excepiton.

Can you confirm?

**Comment by adamvaul** [comment]
I am experienceing a similar issue with this ics data

BEGIN:VCALENDAR
PRODID:-//Syncfusion Inc//Scheduler//EN
VERSION:2.0
CALSCALE:GREGORIAN
METHOD:PUBLISH
X-WR-CALNAME:Calendar
X-WR-TIMEZONE:America/Los_Angeles
BEGIN:VEVENT
LOCATION:
SUMMARY:Event 1
UID:1
Categories:
RRULE:FREQ=WEEKLY;BYDAY=MO,TU,WE,TH,FR;INTERVAL=1;
DTSTART;TZID="America/Los_Angeles":20220802T160000
DTEND;TZID="America/Los_Angeles":20220802T170000
DESCRIPTION:
ISREADONLY:false
END:VEVENT
END:VCALENDAR

When I try to Load a calendar.  It is the same issue as above.

Please

**Comment by minichma** [comment]
Fixed in #645
